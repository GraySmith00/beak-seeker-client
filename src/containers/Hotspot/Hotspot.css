.hotspot {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 2rem;
  height: 100%;
  background: url("./birdBackground4.jpg");
  background-size: cover;
  background-position: top center; }
  .hotspot .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 80px;
    overflow: scroll;
    width: 80%; }
  .hotspot .birds {
    width: 100%; }
  .hotspot .bird {
    display: flex;
    justify-content: space-between;
    margin: 0.5rem auto;
    padding: 0.5rem;
    width: 100%;
    background: rgba(0, 0, 0, 0.75);
    border-radius: 5px;
    -webkit-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43); }
  .hotspot .bird-link {
    text-decoration: none;
    color: #fff; }
  .hotspot .twitter-button {
    padding: 1rem;
    text-decoration: none;
    margin-top: 1rem;
    width: 100%;
    text-align: center;
    border-radius: 5px;
    color: #fff;
    background: #00c6ff;
    /* fallback for old browsers */
    background: -webkit-linear-gradient(to right, #0072ff, #00c6ff);
    /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(to right, #0072ff, #00c6ff);
    /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    -webkit-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43); }
