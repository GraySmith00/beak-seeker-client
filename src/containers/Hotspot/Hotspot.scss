.hotspot {
  display: flex;
  flex-direction: column;
  align-items: center;

  .main-content {
    margin: 60px auto;
  }
}
