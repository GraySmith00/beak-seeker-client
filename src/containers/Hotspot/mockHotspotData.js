export const mockHotspot = {
  locId: 'L1743685',
  locName: 'Belmar Park',
  countryCode: 'US',
  subnational1Code: 'US-CO',
  subnational2Code: 'US-CO-059',
  lat: 39.7044793,
  lng: -105.0857735,
  locID: 'L1743685',
  birds: [
    {
      speciesCode: 'cangoo',
      comName: 'Canada Goose',
      sciName: 'Branta canadensis',
      locId: 'L1743685',
      locName: 'Belmar Park',
      obsDt: '2018-09-09 12:15',
      howMany: 10,
      lat: 39.7044793,
      lng: -105.0857735,
      obsValid: true,
      obsReviewed: false,
      locationPrivate: false
    },
    {
      speciesCode: 'gragoo4',
      comName: 'Graylag Goose (Domestic type) x Canada Goose (hybrid)',
      sciName: 'Anser anser (Domestic type) x Branta canadensis',
      locId: 'L1743685',
      locName: 'Belmar Park',
      obsDt: '2018-09-09 12:15',
      howMany: 1,
      lat: 39.7044793,
      lng: -105.0857735,
      obsValid: true,
      obsReviewed: false,
      locationPrivate: false
    },
    {
      speciesCode: 'mallar3',
      comName: 'Mallard',
      sciName: 'Anas platyrhynchos',
      locId: 'L1743685',
      locName: 'Belmar Park',
      obsDt: '2018-09-09 12:15',
      howMany: 14,
      lat: 39.7044793,
      lng: -105.0857735,
      obsValid: true,
      obsReviewed: false,
      locationPrivate: false
    }
  ]
};

export const mockHotspots = [
  {
    locId: 'L1743685',
    locName: 'Belmar Park',
    countryCode: 'US',
    subnational1Code: 'US-CO',
    subnational2Code: 'US-CO-059',
    lat: 39.7044793,
    lng: -105.0857735,
    locID: 'L1743685',
    birds: [
      {
        speciesCode: 'cangoo',
        comName: 'Canada Goose',
        sciName: 'Branta canadensis',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 10,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      },
      {
        speciesCode: 'gragoo4',
        comName: 'Graylag Goose (Domestic type) x Canada Goose (hybrid)',
        sciName: 'Anser anser (Domestic type) x Branta canadensis',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 1,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      },
      {
        speciesCode: 'mallar3',
        comName: 'Mallard',
        sciName: 'Anas platyrhynchos',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 14,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      },
      {
        speciesCode: 'ribgul',
        comName: 'Ring-billed Gull',
        sciName: 'Larus delawarensis',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 2,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      },
      {
        speciesCode: 'doccor',
        comName: 'Double-crested Cormorant',
        sciName: 'Phalacrocorax auritus',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 4,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      },
      {
        speciesCode: 'grbher3',
        comName: 'Great Blue Heron',
        sciName: 'Ardea herodias',
        locId: 'L1743685',
        locName: 'Belmar Park',
        obsDt: '2018-09-09 12:15',
        howMany: 1,
        lat: 39.7044793,
        lng: -105.0857735,
        obsValid: true,
        obsReviewed: false,
        locationPrivate: false
      }
    ]
  }
];

export const mockCurrentUser = {
  sightings: [
    { _id: 1537200989704, locationId: 'L1743685', speciesCode: 'norsho' }
  ],
  _id: '5b9fd33136b3ce3cd934d36c',
  username: 'PLady1224',
  image:
    'https://pbs.twimg.com/profile_images/1041718097172811776/59BwnDG__normal.jpg',
  __v: 0
};
