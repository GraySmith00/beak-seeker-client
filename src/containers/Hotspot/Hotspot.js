import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { toggleBirdSighting } from '../../actions/thunks/toggleBirdSighting';

import Header from '../../components/Header/Header';
import LoadingSpinner from '../../components/LoadingSpinner/LoadingSpinner';

import './Hotspot.css';

export class Hotspot extends Component {
  constructor() {
    super();
    this.state = {
      hotspot: {},
      loading: true
    };
  }

  componentDidMount() {
    const { hotspotId, currentUser } = this.props;
    const hotspot = this.props.hotspots.find(
      hotspot => hotspot.locId === hotspotId
    );

    this.setState({ hotspot, loading: false });
  }

  handleChange = bird => {
    const { hotspotId, currentUser, toggleBirdSighting } = this.props;
    const newSighting = {
      user: currentUser,
      locationId: hotspotId,
      speciesCode: bird.speciesCode,
      comName: bird.comName
    };
    toggleBirdSighting(newSighting);
  };

  render() {
    const { currentUser } = this.props;
    const { hotspot, loading } = this.state;
    const { birds } = this.state.hotspot;
    let displayBirdLinks;

    if (!loading) {
      displayBirdLinks = birds
        .map((bird, index) => {
          let birdSeen = currentUser.sightings.find(
            sighting =>
              sighting.locationId === hotspot.locId &&
              sighting.speciesCode === bird.speciesCode
          );

          birdSeen ? (birdSeen = true) : (birdSeen = false);

          return (
            <div key={`${index}-${bird.speciesCode}`} className="bird">
              <Link
                to={`/hotspots/${hotspot.locId}/${bird.speciesCode}`}
                className="bird-link"
              >
                {bird.comName}
              </Link>
              <input
                onChange={() => this.handleChange(bird)}
                type="checkbox"
                name={bird.speciesCode}
                checked={birdSeen}
                className="checkbox-input"
              />
            </div>
          );
        })
        .slice(0, 10);
    }

    return (
      <div className="hotspot">
        <Header currentPage="Hotspot Info" />
        <main className="main-content">
          <h2>{this.state.hotspot.locName}</h2>
          <div className="birds">
            <form>{displayBirdLinks}</form>
          </div>
          <Link to={`/tweet`} className="twitter-button">
            Tweet Recent Sighting!
          </Link>
        </main>
      </div>
    );
  }
}

Hotspot.propTypes = {
  hotspotId: PropTypes.string.isRequired,
  hotspots: PropTypes.array.isRequired,
  currentUser: PropTypes.object,
  toggleBirdSighting: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  hotspots: state.hotspots,
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  toggleBirdSighting: (user, locationId, speciesCode) =>
    dispatch(toggleBirdSighting(user, locationId, speciesCode))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Hotspot);
