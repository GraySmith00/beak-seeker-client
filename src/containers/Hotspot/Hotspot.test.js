import React from 'react';
import { shallow } from 'enzyme';
import { Hotspot } from './Hotspot';
import { mockHotspot, mockHotspots } from './mockHotspotData';

describe('Hotspot component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Hotspot hotspotId={mockHotspot.locId} hotspots={mockHotspots} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
