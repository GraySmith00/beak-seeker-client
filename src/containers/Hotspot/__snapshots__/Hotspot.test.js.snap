// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotspot component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hotspot
    hotspotId="L1743685"
    hotspots={
        Array [
            Object {
              "birds": Array [
                Object {
                  "comName": "Canada Goose",
                  "howMany": 10,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Branta canadensis",
                  "speciesCode": "cangoo",
                },
                Object {
                  "comName": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
                  "howMany": 1,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Anser anser (Domestic type) x Branta canadensis",
                  "speciesCode": "gragoo4",
                },
                Object {
                  "comName": "Mallard",
                  "howMany": 14,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Anas platyrhynchos",
                  "speciesCode": "mallar3",
                },
                Object {
                  "comName": "Ring-billed Gull",
                  "howMany": 2,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Larus delawarensis",
                  "speciesCode": "ribgul",
                },
                Object {
                  "comName": "Double-crested Cormorant",
                  "howMany": 4,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Phalacrocorax auritus",
                  "speciesCode": "doccor",
                },
                Object {
                  "comName": "Great Blue Heron",
                  "howMany": 1,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Ardea herodias",
                  "speciesCode": "grbher3",
                },
              ],
              "countryCode": "US",
              "lat": 39.7044793,
              "lng": -105.0857735,
              "locID": "L1743685",
              "locId": "L1743685",
              "locName": "Belmar Park",
              "subnational1Code": "US-CO",
              "subnational2Code": "US-CO-059",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Belmar Park
</h1>,
        <div
          className="birds"
>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/cangoo"
                    >
                              Canada Goose
                    </Link>
          </div>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/gragoo4"
                    >
                              Graylag Goose (Domestic type) x Canada Goose (hybrid)
                    </Link>
          </div>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/mallar3"
                    >
                              Mallard
                    </Link>
          </div>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/ribgul"
                    >
                              Ring-billed Gull
                    </Link>
          </div>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/doccor"
                    >
                              Double-crested Cormorant
                    </Link>
          </div>
          <div
                    className="bird"
          >
                    <Link
                              replace={false}
                              to="/hotspots/L1743685/grbher3"
                    >
                              Great Blue Heron
                    </Link>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Belmar Park",
        },
        "ref": null,
        "rendered": "Belmar Park",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/cangoo"
              >
                            Canada Goose
              </Link>
</div>,
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/gragoo4"
              >
                            Graylag Goose (Domestic type) x Canada Goose (hybrid)
              </Link>
</div>,
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/mallar3"
              >
                            Mallard
              </Link>
</div>,
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/ribgul"
              >
                            Ring-billed Gull
              </Link>
</div>,
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/doccor"
              >
                            Double-crested Cormorant
              </Link>
</div>,
            <div
              className="bird"
>
              <Link
                            replace={false}
                            to="/hotspots/L1743685/grbher3"
              >
                            Great Blue Heron
              </Link>
</div>,
          ],
          "className": "birds",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0-cangoo",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/cangoo"
>
                Canada Goose
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Canada Goose",
                "replace": false,
                "to": "/hotspots/L1743685/cangoo",
              },
              "ref": null,
              "rendered": "Canada Goose",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1-gragoo4",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/gragoo4"
>
                Graylag Goose (Domestic type) x Canada Goose (hybrid)
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
                "replace": false,
                "to": "/hotspots/L1743685/gragoo4",
              },
              "ref": null,
              "rendered": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2-mallar3",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/mallar3"
>
                Mallard
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Mallard",
                "replace": false,
                "to": "/hotspots/L1743685/mallar3",
              },
              "ref": null,
              "rendered": "Mallard",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "3-ribgul",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/ribgul"
>
                Ring-billed Gull
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ring-billed Gull",
                "replace": false,
                "to": "/hotspots/L1743685/ribgul",
              },
              "ref": null,
              "rendered": "Ring-billed Gull",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "4-doccor",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/doccor"
>
                Double-crested Cormorant
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Double-crested Cormorant",
                "replace": false,
                "to": "/hotspots/L1743685/doccor",
              },
              "ref": null,
              "rendered": "Double-crested Cormorant",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "5-grbher3",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/hotspots/L1743685/grbher3"
>
                Great Blue Heron
</Link>,
              "className": "bird",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Great Blue Heron",
                "replace": false,
                "to": "/hotspots/L1743685/grbher3",
              },
              "ref": null,
              "rendered": "Great Blue Heron",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Belmar Park
</h1>,
          <div
            className="birds"
>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/cangoo"
                        >
                                    Canada Goose
                        </Link>
            </div>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/gragoo4"
                        >
                                    Graylag Goose (Domestic type) x Canada Goose (hybrid)
                        </Link>
            </div>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/mallar3"
                        >
                                    Mallard
                        </Link>
            </div>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/ribgul"
                        >
                                    Ring-billed Gull
                        </Link>
            </div>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/doccor"
                        >
                                    Double-crested Cormorant
                        </Link>
            </div>
            <div
                        className="bird"
            >
                        <Link
                                    replace={false}
                                    to="/hotspots/L1743685/grbher3"
                        >
                                    Great Blue Heron
                        </Link>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Belmar Park",
          },
          "ref": null,
          "rendered": "Belmar Park",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/cangoo"
                >
                                Canada Goose
                </Link>
</div>,
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/gragoo4"
                >
                                Graylag Goose (Domestic type) x Canada Goose (hybrid)
                </Link>
</div>,
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/mallar3"
                >
                                Mallard
                </Link>
</div>,
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/ribgul"
                >
                                Ring-billed Gull
                </Link>
</div>,
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/doccor"
                >
                                Double-crested Cormorant
                </Link>
</div>,
              <div
                className="bird"
>
                <Link
                                replace={false}
                                to="/hotspots/L1743685/grbher3"
                >
                                Great Blue Heron
                </Link>
</div>,
            ],
            "className": "birds",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0-cangoo",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/cangoo"
>
                  Canada Goose
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Canada Goose",
                  "replace": false,
                  "to": "/hotspots/L1743685/cangoo",
                },
                "ref": null,
                "rendered": "Canada Goose",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1-gragoo4",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/gragoo4"
>
                  Graylag Goose (Domestic type) x Canada Goose (hybrid)
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
                  "replace": false,
                  "to": "/hotspots/L1743685/gragoo4",
                },
                "ref": null,
                "rendered": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2-mallar3",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/mallar3"
>
                  Mallard
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Mallard",
                  "replace": false,
                  "to": "/hotspots/L1743685/mallar3",
                },
                "ref": null,
                "rendered": "Mallard",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "3-ribgul",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/ribgul"
>
                  Ring-billed Gull
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ring-billed Gull",
                  "replace": false,
                  "to": "/hotspots/L1743685/ribgul",
                },
                "ref": null,
                "rendered": "Ring-billed Gull",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "4-doccor",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/doccor"
>
                  Double-crested Cormorant
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Double-crested Cormorant",
                  "replace": false,
                  "to": "/hotspots/L1743685/doccor",
                },
                "ref": null,
                "rendered": "Double-crested Cormorant",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "5-grbher3",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/hotspots/L1743685/grbher3"
>
                  Great Blue Heron
</Link>,
                "className": "bird",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Great Blue Heron",
                  "replace": false,
                  "to": "/hotspots/L1743685/grbher3",
                },
                "ref": null,
                "rendered": "Great Blue Heron",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
